neuraxle/union.py

```diff
@@ -87,110 +64,68 @@ def _fit_data_container(self, data_container, context):
         if self.n_jobs != 1:
             fitted_steps = Parallel(backend=self.backend, n_jobs=self.n_jobs)(
                 delayed(step.handle_fit)(data_container.copy(), context)
-                for _, step in self.steps_as_tuple
+                for _, step in self.steps_as_tuple[:-1]
             )
         else:
             fitted_steps = [
                 step.handle_fit(data_container.copy(), context)
-                for _, step in self.steps_as_tuple
+                for _, step in self.steps_as_tuple[:-1]
             ]
 
-        # Save fitted steps
-        for i, fitted_step in enumerate(fitted_steps):
-            self.steps_as_tuple[i] = (self.steps_as_tuple[i][0], fitted_step)
-        self._refresh_steps()
+        self._save_fitted_steps(fitted_steps)
 
         return self
 
     def _transform_data_container(self, data_container, context):
         """
         Transform the data with the unions. It will make use of some parallel processing.
-
         :param data_container: data container
         :param context: execution context
         :return: the transformed data_inputs.
         """
         if self.n_jobs != 1:
             data_containers = Parallel(backend=self.backend, n_jobs=self.n_jobs)(
                 delayed(step.handle_transform)(data_container.copy(), context)
-                for _, step in self.steps_as_tuple
+                for _, step in self.steps_as_tuple[:-1]
             )
         else:
             data_containers = [
                 step.handle_transform(data_container.copy(), context)
-                for _, step in self.steps_as_tuple
+                for _, step in self.steps_as_tuple[:-1]
             ]
 
-        return DataContainer(data_inputs=data_containers, current_ids=data_container.current_ids,
-                             summary_id=data_container.summary_id, expected_outputs=data_container.expected_outputs)
+        return DataContainer(
+            data_inputs=data_containers,
+            current_ids=data_container.current_ids,
+            summary_id=data_container.summary_id,
+            expected_outputs=data_container.expected_outputs
+        )
 
     def _did_transform(self, data_container, context):
-        data_container = self.joiner.handle_transform(data_container, context)
+        data_container = self[-1].handle_transform(data_container, context)
```

Comment:
would prefer self.get_joiner() instead of an insidious -1, although I understand. 
(optional fix, no time for that).